internal/tests_config.go

```diff
@@ -57,35 +59,93 @@ type Stats struct {
 //
 // May contain prefixes; the longest prefix wins.
 type TestsConfig struct {
-	Default status   `yaml:"default"`
-	Stats   *Stats   `yaml:"stats"`
-	Pass    []string `yaml:"pass"`
-	Skip    []string `yaml:"skip"`
-	Fail    []string `yaml:"fail"`
+	Default status
+	Stats   *Stats
+	Pass    Tests
+	Skip    Tests
+	Fail    Tests
 }
 
-func (tc *TestsConfig) toMap() (map[string]status, error) {
-	res := make(map[string]status, len(tc.Pass)+len(tc.Skip)+len(tc.Fail))
+type Tests struct {
+	TestNames []string
+	ResRegexp []string
+}
 
-	for _, t := range tc.Pass {
-		if _, ok := res[t]; ok {
-			return nil, fmt.Errorf("duplicate test or prefix: %q", t)
-		}
-		res[t] = Pass
+type ImportTestsConfig struct {
+	Default status `yaml:"default"`
+	Stats   *Stats `yaml:"stats"`
+	Pass    []any  `yaml:"pass"`
+	Skip    []any  `yaml:"skip"`
+	Fail    []any  `yaml:"fail"`
+}
+
+func (itc *ImportTestsConfig) Convert() (*TestsConfig, error) {
+	if itc == nil {
+		return nil, nil // not sure if that works
 	}
+	tc := TestsConfig{itc.Default, itc.Stats, Tests{}, Tests{}, Tests{}}
+	for _, tcat := range []struct {
+		inTests  []any
+		outTests *Tests
+	}{
+		{itc.Pass, &tc.Pass},
+		{itc.Skip, &tc.Skip},
+		{itc.Fail, &tc.Fail},
+	} {
+		for _, t := range tcat.inTests {
+			switch test := t.(type) {
+			case map[string]any:
+				//if m, ok := t.(map[string]any); ok {
+				mValue, ok := test["regexp"]
+				if !ok {
+					return nil, fmt.Errorf("invalid field name (\"regexp\" expected)")
+				}
+				regexp, ok := mValue.(string)
+				if !ok {
+					// Check specifically for an array
+					if _, ok := mValue.([]string); ok {
+						return nil, fmt.Errorf("invalid syntax: regexp value shouldn't be an array")
+					}
+					return nil, fmt.Errorf("invalid syntax: expected string, got: %v", reflect.TypeOf(mValue))
+				}
 
-	for _, t := range tc.Skip {
-		if _, ok := res[t]; ok {
-			return nil, fmt.Errorf("duplicate test or prefix: %q", t)
+				tcat.outTests.ResRegexp = append(tcat.outTests.ResRegexp, regexp)
+				continue
+				//}
+			case string:
+				//if testname, ok := t.(string); ok {
+				tcat.outTests.TestNames = append(tcat.outTests.TestNames, test)
+				continue
+				//}
+			default:
+				return nil, fmt.Errorf("invalid type of \"%q\": %q", t, reflect.TypeOf(t))
```

Comment:
```suggestion
				return nil, fmt.Errorf("invalid type of %[1]q: %[1]T", t)
```

`%T` does what we need, and `%q` adds quotes